### DISCUSSION TITLE
关于项目中场景类的析构函数使用virtual前缀的问题

### DISCUSSION THREAD

**Discussion by robin-bird-go** [mayerui/sudoku#27] [General]
您好，我也在研究您的数独游戏源码内容，然后看到CScence的类的析构函数前面添加了virtual前缀，但是不知道有什么用途。（一般自己把出现虚函数的地方给认为是会有一些派生的子类实现多态的，但是这个场景类好像没有子类？）

（顺便给项目内容点个赞，学到很多）

**Comment by mayerui** [comment]
目前CScence还没有涉及继承，但设计上并没有禁止继承，也许以后会实现NxN的棋盘（当然这不一定非要用子类）或是更多样规则的场景？一旦涉及继承，最需要关注的问题之一就是析构函数，这严格来说可能还无关多态。如果父类的析构函数不是虚函数的话，那么子类在析构的过程中就不会调用父类析构，会有资源泄漏风险。
